micras_proxy/src/wall_sensors.cpp

```diff
@@ -42,18 +41,18 @@ void TWallSensors<num_of_sensors>::update() {
 }
 
 template <uint8_t num_of_sensors>
-core::Observation TWallSensors<num_of_sensors>::get_observation(uint8_t sensor_index) const {
-    float reading = this->filters.at(sensor_index).get_last();
-
-    if (reading > this->wall_threshold.at(sensor_index)) {
-        return core::Observation::WALL;
-    }
+bool TWallSensors<num_of_sensors>::get_wall(uint8_t sensor_index, bool disturbed) const {
+    return this->filters.at(sensor_index).get_last() >
+           this->base_readings.at(sensor_index) * this->uncertainty * (disturbed ? 1.2F : 1.0F);
+}
 
-    if (reading < this->free_space_threshold.at(sensor_index)) {
-        return core::Observation::FREE_SPACE;
+template <uint8_t num_of_sensors>
+core::Observation TWallSensors<num_of_sensors>::get_observation() const {
+    if (this->get_wall(0) && this->get_wall(3)) {
+        return {this->get_wall(1, true), true, this->get_wall(2, true)};
```

Comment:
```suggestion
template <uint8_t num_of_sensors>
core::Observation TWallSensors<num_of_sensors>::get_observation() const {
    if (this->get_wall(0) && this->get_wall(3)) {
        return {
            .left = this->get_wall(1, true),
            .front = true,
            .right = this->get_wall(2, true)
        };
```

Mano acho que podia retornar assim, fica mai legivel e claro pra ver oq é cada coisa do return